### PR TITLE
feat: allow merging of keymap preset with custom keymap

### PR DESCRIPTION
**Pull request by kedom1337**
This PR adds the ability to merge a custom keymap with one of the keymap presets.
This is achieved by

1. Extending the `keymap` configuration with a `preset` key which specifies which preset to use
2. Using `vim.tbl_extend` to then merge the preset with the custom keymap

Example:
```lua
keymap = {
  preset = 'default',
  ["<cr>"] = { "select_and_accept", "fallback" },
},
```

### CODE REVIEW

File: lua/blink/cmp/keymap.lua [hunk 1]
```diff
@@ -71,6 +71,25 @@ function keymap.setup(opts)
...
+    -- Handle preset inside table
+    if opts.preset then
+      local preset_keymap
+      if opts.preset == 'default' then
+        preset_keymap = default_keymap
+      elseif opts.preset == 'super-tab' then
+        preset_keymap = super_tab_keymap
+      else
+        error('Invalid blink.cmp keymap preset: ' .. opts.preset)
+      end
```
Comment (Saghen) [inline lua/blink/cmp/keymap.lua:84]: Move this to a separate `keymap.get_preset()`

**Review by Saghen — COMMENTED** [review]
Looks great thanks! Some conflicts on the readme since I just added an `enter` preset and one minor change

**Comment by Saghen** [comment]
Thanks again!

**Comment by kedom1337** [comment]
I should thank you :). Thanks for the awesome plugin!

**Pull request merged**